public/request/achievement/update-base.php

```diff
@@ -26,6 +27,15 @@
     abort(403);
 }
 
+// Don't allow adding types to subsets or test kits.
+$game = Game::find((int) $achievement['GameID']);
+if ($game) {
+    $canHaveTypes = mb_strpos($game->Title, "[Subset") === false && mb_strpos($game->Title, "~Test Kit~") === false;
```

Comment:
Recommend extending this to Events.

This logic should probably be in a function somewhere - maybe on the `Game` model?

Reply:
Agreed - moved to `Game` as `getCanHaveTypes()`.